src/ring_theory/finiteness.lean

```diff
@@ -401,6 +442,17 @@ by { rw ← f.comp_id, exact (id A).comp_surjective hf hker}
 lemma of_finite_type [is_noetherian_ring A] {f : A →+* B} : f.finite_type ↔ f.finite_presentation :=
 @algebra.finite_presentation.of_finite_type A B _ _ f.to_algebra _
 
+lemma comp {g : B →+* C} {f : A →+* B} (hg : g.finite_presentation) (hf : f.finite_presentation) :
+  (g.comp f).finite_presentation :=
+@algebra.finite_presentation.trans A B C _ _ f.to_algebra _ (g.comp f).to_algebra g.to_algebra
+begin
+  fconstructor,
+  intros a b c,
+  simp only [algebra.smul_def, ring_hom.map_mul, mul_assoc],
+  refl
+end
```

Comment:
```suggestion
{ smul_assoc := λ a b c, begin
    simp only [algebra.smul_def, ring_hom.map_mul, mul_assoc],
    refl
  end }
```
makes it a little clearer what exactly is being proved here

Reply:
If this form is preferred I can modify the other proofs where it is used (it is the same for finitely generated/finite).